apps/core/src/hooks/useGetFiatBalance.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useCurrentAccount } from '@iota/dapp-kit';
+import { useBalanceInUSD } from './useTokenPrice';
+import { IOTA_TYPE_ARG } from '@iota/iota-sdk/utils';
+import { useBalance } from './useBalance';
+import { formatBalanceToUSD } from '../utils';
+import { Network } from '@iota/iota-sdk/client';
+
+export function useGetFiatBalance(network: Network): string | null {
+    const account = useCurrentAccount();
+    const address = account?.address;
+    if (!address) return null;
```

Comment:
This could very easily to the classic "react has detected a change in the order of hooks" error, I would remove this early return